sky/clouds/service_catalog/azure_catalog.py

```diff
@@ -68,7 +68,6 @@ def get_hourly_cost(instance_type: str,
                     region: Optional[str] = None,
                     zone: Optional[str] = None) -> float:
     # Ref: https://azure.microsoft.com/en-us/support/legal/offer-details/
-    assert not use_spot, 'Current Azure subscription does not support spot.'
```

Comment:
Do you think we should print a warning or a hint that the spot feature on Azure may not be well tested and let the user to file an issue on GitHub?

Reply:
Feels like it should work. Adding may give a sense of being experimental to users. Wdyt?

Reply:
Sounds good. Approving this PR for now. Let's wait and see any feedback from users.